### PR TITLE
Wiki history

### PR DESCRIPTION
**Pull request by AshakaE**
# New Entity(Activity) for storing wiki history

_PR description goes here_

## How should this be tested?

1. ### Activites 

> `query {
  Activities {
    datetime
    ipfs
    content {
      id
      ipfs
    }
  }
}`

2. ### ActivitiesByWikiId
> `query {
  ActivitiesByWikiId(wikiId: "Wiki title") {
    datetime
    ipfs
    content {
      id
      ipfs
    }
  }
}`

## Notes or observations

_jot down something here_

## Linked issues

closes #121, closes https://github.com/EveripediaNetwork/issues/issues/121

### CODE REVIEW

File: src/Database/Entities/activity.entity.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  @Field(() => ID)
+  @PrimaryGeneratedColumn('increment')
+  id!: string
+
+  @Field()
+  @Column('varchar')
+  wikiId!: string
+
+  @Field(() => Status)
+  @Column('enum', { enum: Status, nullable: true })
```
Comment (kesar) [inline src/Database/Entities/activity.entity.ts:37]: this shouldnt be nullable

File: src/Database/Entities/activity.entity.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  type!: Status
+
+  @Field(() => [Wiki])
+  @Column('jsonb', { nullable: true })
```
Comment (kesar) [inline src/Database/Entities/activity.entity.ts:41]: this shouldnt be nullable

File: src/App/activity.resolver.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  @Field(() => String)
+  wikiId!: string
+}
+
+@Resolver(() => Activity)
+class ActivityResolver {
+  constructor(private connection: Connection) {}
+
+  @Query(() => [Activity])
+  async Activities(@Args() args: PaginationArgs) {
```
Comment (kesar) [inline src/App/activity.resolver.ts:17]: something I realized now (not in the specs) its that we might want to have the language filter here like we do in wikis, defaulting to en (copy the same idea of the other resolver pls)

File: src/Indexer/Store/store.service.ts [hunk 4]
```diff
@@ -84,9 +86,36 @@ class DBStoreService {
...
+        ],
+        datetime: new Date(Date.now()),
+        ipfs: hash.id,
+      })
+      return resp
+    }
 
     // TODO: store history and delete?
     if (existWiki) {
+      await activityRepository.save(createActivity(Status.UPDATED))
```
[the remark this answers is not in the document]
Comment (AshakaE) [reply src/Indexer/Store/store.service.ts:118]: Yes, this works fine. I actually had an issue with the status(type), it was initially not getting the first wiki creation(new Wiki), but what this does is it adds the first wiki(on wiki create) then adds each other update( on wiki update). So for each wiki in the activity table gets a status(created) except it's updated and each update gets stored as well

**Review by kesar — APPROVED** [review]

**Review by kesar — APPROVED** [review]

**Review by kesar — APPROVED** [review]

**Pull request merged**